src/transformers/generation_logits_process.py

```diff
@@ -702,3 +702,31 @@ class LogitNormalization(LogitsProcessor, LogitsWarper):
     def __call__(self, input_ids: torch.Tensor, scores: torch.Tensor) -> torch.Tensor:
         scores = scores.log_softmax(dim=-1)
         return scores
+
+
+class SuppressBlank(LogitsProcessor):
```

Comment:
Actually we should make this more general to be useable by more models.
Could we change this logit processor to `SuppressTokensAtBegin` where we pass all tokens in a generic way that should be suppressed, which in this case would be `[blank_token_id, eos_token_id]` . The user should not be concerned with having to pass `sample_begin`. We can automatically figure out this value in `generate`